extensions/common/iam/verifiable-credentials/src/main/java/org/eclipse/edc/iam/verifiablecredentials/StatusList2021RevocationService.java

```diff
@@ -55,7 +60,45 @@ public Result<Void> checkValidity(VerifiableCredential credential) {
                 .orElse(Result.failure("Could not check the validity of the credential with ID '%s'".formatted(credential.getId())));
     }
 
+    @Override
+    public Result<String> getStatusPurpose(VerifiableCredential credential) {
+        if (credential.getCredentialStatus().isEmpty()) {
+            return successNullable(null);
+        }
```

Comment:
this guard is not necessary, when the status is empty the method will return a success(null) anyway

Reply:
it's either the guard here, or I refactor or add a similar check in Line 76:
```java
 var list = res.get(true).stream()
                .filter(r -> r.getContent() != null)
                .map(AbstractResult::getContent).toList();
```

Reply:
nope, because the stream will be empty so the filter function will never be evaluated

Reply:
yes, but the `.get(true)` will, which is what'll cause the NPE

Reply:
ouch!